### PR TITLE
CRW-1093 copy downstream Dockerfile to...

### PR DESCRIPTION
**Pull request by nickboldt**
CRW-1093 copy downstream Dockerfile to rhel.Dockerfile upstream so we can better sync che server w/ crw server

Change-Id: Id76c6c38c4e17708e577de8abc5acde4b8376edf
Signed-off-by: nickboldt <nboldt@redhat.com>

### CODE REVIEW

**Comment by nickboldt** [comment]
Once merged, we can finally remove https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/crw_sync-github-to-pkgs.devel/ and merge its functionality into https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/crw_master/ via:

* https://github.com/redhat-developer/codeready-workspaces/pull/325
* https://github.com/redhat-developer/codeready-workspaces/blob/master/Jenkinsfile#L221

**Comment by nickboldt** [comment]
[crw-ci-test]

**Comment by nickboldt** [comment]
I love how creating a new file with no dependencies on existing code can cause unhappy path to fail :(

**Comment by nickboldt** [comment]
Duped for 7.16.x branch in https://github.com/eclipse/che/pull/17496

File: dockerfiles/che/rhel.Dockerfile
Comment (skabashnyuk) [inline dockerfiles/che/rhel.Dockerfile]: That is something that does not exist in upstream. I suggest having one more Dockerfile in https://github.com/redhat-developer/codeready-workspaces-productization that overrides this content as an additional layer.

**Comment by skabashnyuk** [comment]
@nickboldt from what I can see downstream changes only bringing https://github.com/redhat-developer/codeready-workspaces/tree/master/assembly/branding. Can we make it based on upstream image where breeding added as an extra layer? In this way, we will get a fully functional upstream `ubi8` image.

**Review by amisevsk — COMMENTED** [review]
I'm not sure where we land on having prod-specific bits in this repo, but I wanted to mention @sleshchenko's pretty elegant solution to having an upstream/downstream split in a single dockerfile: we have a [template](https://github.com/redhat-developer/web-terminal-tooling/blob/v1.0.x/build/template.Dockerfile) that we use to [generate](https://github.com/redhat-developer/web-terminal-tooling/blob/v1.0.x/build/generate_dockerfile.sh) the dockerfile variations with a single `sed` replacement.

**Comment by nickboldt** [comment]
> I'm not sure where we land on having prod-specific bits in this repo

Generally I get crapped on for trying to put anything brew-ish into community projects.  web-terminal is downstream (midstream) so it can blur the lines a little. I like the templated approach @amisevsk - similar to my approach where I assume a Brew friendly script and then apply local transforms to make it run outside brew.  @sleshchenko's is cleaner, though. :D 

@skabashnyuk do you object to my solution? Or are you suggesting a future enhancement to get CRW closer to upstream? 

Right now I'm just focused on CRW-1093 -- preventing the entrypoint from getting outdated. Can we merge this and https://github.com/eclipse/che/pull/17496 so I can verify it works in Jenkins and Brew?

File: dockerfiles/che/rhel.Dockerfile
Comment (nickboldt) [reply dockerfiles/che/rhel.Dockerfile:33]: So you're suggesting that I should have a `COPY` line that does something with locally built `che` bits, and when reusing it downstream, I'd transform it via sed to `COPY` in the CRW bits instead? 

Seems like we're making things more complicated than they need to be, but you're right, the crw tarball isn't in the upstream repo.

Comment (nickboldt) [reply dockerfiles/che/rhel.Dockerfile]: True, in upstream it's just:
```
ADD eclipse-che /home/user/eclipse-che
```
https://github.com/eclipse/che/blob/master/dockerfiles/che/Dockerfile#L29

But in https://github.com/redhat-developer/codeready-workspaces-productization we have only process docs. We could transform an upstream rhel.Dockerfile into downstream https://github.com/redhat-developer/codeready-workspaces/blob/master/Dockerfile via https://github.com/redhat-developer/codeready-workspaces/pull/325/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R317-R322 by adding some new rules (or moving the whole thing to use @sleshchenko 's templated approach, but that requires having references to Brew in upstream)

**Review by amisevsk — APPROVED** [review]

**Comment by benoitf** [comment]
how could it be merged ?
No code owners approval

**Comment by nickboldt** [comment]
> amisevsk approved these changes 13 minutes ago

**Comment by benoitf** [comment]
Afaik Angel is not code owner of this module

**Comment by nickboldt** [comment]
Are you suggesting that we want to limit commits to being ONLY allowed by code owners?

This was approved conceptually by Mario (in CRW-1093) and Sergii K (above)

https://issues.redhat.com/browse/CRW-1093

**Comment by benoitf** [comment]
commits from everyone, and then at least one code owner approval ( not on concepts)
This is why there are code owners

**Pull request merged**
